enterprise/server/mockoauth/mockoauth.go

```diff
@@ -0,0 +1,253 @@
+package mockoauth
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"math/big"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/config"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/lestrrat-go/jwx/jwa"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/lestrrat-go/jwx/jwt"
+	"golang.org/x/oauth2"
+)
+
+func Provider(env environment.Env) config.OauthProvider {
+	return config.OauthProvider{
+		IssuerURL:    env.GetConfigurator().GetAppBuildBuddyURL(),
+		ClientID:     "buildbuddy",
+		ClientSecret: "secret",
+	}
+}
+
+type oauth2Mock struct {
+	env           environment.Env
+	rsaPrivateKey jwk.RSAPrivateKey
+	rsaPublicKey  jwk.RSAPublicKey
+}
+
+type configurationJSON struct {
+	Issuer                           string   `json:"issuer"`
+	AuthorizationEndpoint            string   `json:"authorization_endpoint"`
+	TokenEndpoint                    string   `json:"token_endpoint"`
+	UserinfoEndpoint                 string   `json:"userinfo_endpoint"`
+	JwksUri                          string   `json:"jwks_uri"`
+	IdTokenSigningAlgValuesSupported []string `json:"id_token_signing_alg_values_supported"`
+}
+
+type tokenJSON struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresIn    int32  `json:"expires_in"`
+	IdToken      string `json:"id_token"`
+}
+
+func NewMockOauth(env environment.Env) (*oauth2Mock, error) {
+	// generate the same key every time; this is not meant to secure or secret
+	var n, d, p, q big.Int
+	n.SetString(
+		"23854054546089486752143428681593425828175873941670684122390466251269713159"+
+			"245737985667367468143300414391912523520647401288672254625181245443791922"+
+			"131328655375075086232375550780596731379873494676833765863924320065467041"+
+			"534880370068444461206120370979494670081178693685915972335443995410513383"+
+			"318660804108686793740105181807521526108041735581000243371902164278468459"+
+			"474539325728897031680901033148357912833321597059542124576712419669819030"+
+			"691795237787947940066997209320864327657065503764510504770118773928022493"+
+			"125524472354132671270720851491074770807035708210053594371353758234068435"+
+			"932384388342343398119337427425316157863",
```

Comment:
to do a raw string in go, you can use backticks (same below)

Reply:
I think the idiomatic way to do this in go is to use a raw string. Here's a specific example of a key in the stdlib (albeit in a test): https://cs.opensource.google/go/go/+/master:src/crypto/x509/x509_test.go;l=106?q=%22Key%20%3D%20%60%22&ss=go%2Fgo

But there are many example of embedding content like this: https://cs.opensource.google/search?q=%22%3D%20%60%22&ss=go%2Fgo (usages, examples, programs, etc.)

Reply:
Sure, but this is a string representing a base 10 number, not a standard string representation of an RSA key. `big.Int` won't be able to parse anything with non-digit (in this case, new-line) characters.

To be more clear, I could use a single raw string with backticks, but it would just look like a single raw string with double quotes:
```
n.SetString(
		`23854054546089486752143428681593425828175873941670684122390466251269713159245737985667367468143300414391912523520647401288672254625181245443791922131328655375075086232375550780596731379873494676833765863924320065467041534880370068444461206120370979494670081178693685915972335443995410513383318660804108686793740105181807521526108041735581000243371902164278468459474539325728897031680901033148357912833321597059542124576712419669819030691795237787947940066997209320864327657065503764510504770118773928022493125524472354132671270720851491074770807035708210053594371353758234068435932384388342343398119337427425316157863`,
		10)
```
and, specifically, it can't be:
```
n.SetString(
		`23854054546089486752143428681593425828175873941670684122390466251269713159
245737985667367468143300414391912523520647401288672254625181245443791922
131328655375075086232375550780596731379873494676833765863924320065467041
534880370068444461206120370979494670081178693685915972335443995410513383
318660804108686793740105181807521526108041735581000243371902164278468459
474539325728897031680901033148357912833321597059542124576712419669819030
691795237787947940066997209320864327657065503764510504770118773928022493
125524472354132671270720851491074770807035708210053594371353758234068435
932384388342343398119337427425316157863`,
		10)
```

because that string has new lines, which `big.Int` can't parse.

Reply:
Can make these things consts at the top (using raw string) and in your code call n.SetString(strings.Join(strings.Fields(myKey), ""))?

Reply:
works for me.